core/templates/services/debouncer.service.ts

```diff
@@ -27,32 +27,28 @@ export class DebouncerService {
   // be invoked. The function only gets executed after it stops being called
   // for `wait` milliseconds.
   debounce(func: () => void, millisecsToWait: number): () => void {
-    let timeout;
-    let context = this;
-    let args = arguments;
-    let timestamp;
-    let result;
+    let timeout: number;
+    let context: this | null = this;
+    let timestamp: number;
+    let result: void;
 
     const later = () => {
       let last = new Date().getTime() - timestamp;
       if (last < millisecsToWait) {
-        timeout = setTimeout(later, millisecsToWait - last);
+        timeout = window.setTimeout(later, millisecsToWait - last);
```

Comment:
@DubeySandeep here Instead of adding window to setTimeout and setting type of timeout to number we can set the type of timeout to NodeJS.timeout i.e ``` let timeout: NodeJS.timeout ```. I currently set it to number because it then we can set the value of timeout later in the function to 0 instead of null.